context/app/static/js/components/detailPage/DatasetRelationships/Legend.tsx

```diff
@@ -0,0 +1,93 @@
+import React, { PropsWithChildren } from 'react';
+import Box from '@mui/material/Box';
+import Typography from '@mui/material/Typography';
+import Stack from '@mui/material/Stack';
+import InfoTooltipIcon from 'js/shared-styles/icons/TooltipIcon';
+import { nodeIcons, nodeColors } from './nodeTypes';
+import StatusIcon from '../StatusIcon';
+
+interface NodeLegendItemProps {
+  name: string;
+  nodeKey: keyof typeof nodeIcons;
+}
+
+function NodeLegendItem({ name, nodeKey }: NodeLegendItemProps) {
+  const Icon = nodeIcons[nodeKey];
+  const borderRadius = nodeKey === 'pipeline' ? 0 : 4;
+  return (
+    <Stack direction="row" gap={0.5} alignItems="center">
+      <Box
+        width={32}
+        height={16}
+        borderRadius={borderRadius}
+        bgcolor={nodeColors[nodeKey]}
+        display="inline-flex"
+        alignItems="center"
+        justifyContent="center"
+      >
+        {Icon && <Icon color="primary" fontSize="0.75rem" width="0.75rem" height="0.75rem" />}
```

Comment:
What necessitated both `fontSize` and `width` + `height`?

Reply:
I believe this was an artifact from troubleshooting the issue with dynamic status icons causing flickering, I'll remove the unnecessary width/height props.